pages/pil/unscoped/courses/read/views/index.jsx

```diff
@@ -60,6 +60,20 @@ const tableFormatters = {
   }
 };
 
+// This check can be removed once all training courses have coursePurpose set
+const checkCoursePurposeBeforeApplyForLicence = (model, canUpdate) => {
+  if (model.coursePurpose) {
+    return canUpdate && <Link page="pils.courses.participants.add" className="govuk-button button-secondary" label={<Snippet>buttons.apply</Snippet>} />;
+  } else {
+    return (
+      <>
+        <Warning><Snippet>coursePurposeRequiredWarning</Snippet></Warning>
+        <Link page="pils.courses.update" className="govuk-button button-secondary" label={<Snippet>buttons.edit</Snippet>} />
+      </>
+    );
+  }
+};
```

Comment:
I think canUpdate should also be checked before displaying the warning and edit button
```suggestion
const checkCoursePurposeBeforeApplyForLicence = (model, canUpdate) => {
  if (!canUpdate) {
    return null;
  }
  else if (model.coursePurpose) {
    return <Link page="pils.courses.participants.add" className="govuk-button button-secondary" label={<Snippet>buttons.apply</Snippet>} />;
  } else {
    return (
      <>
        <Warning><Snippet>coursePurposeRequiredWarning</Snippet></Warning>
        <Link page="pils.courses.update" className="govuk-button button-secondary" label={<Snippet>buttons.edit</Snippet>} />
      </>
    );
  }
};
```

Reply:
I thought about it but I am not 100% sure. Happy to test it with above change. I will push the change now.